lib/core/unittest/CProgramCountersTest.cc

```diff
@@ -65,181 +37,187 @@ void CProgramCountersTest::testCounters() {
     using TCounter = ml::core::CProgramCounters::TCounter;
 
     static const int N = 6;
-    for (int i = 0; i < N; i++) {
+    for (int i = 0; i < N; ++i) {
         CPPUNIT_ASSERT_EQUAL(TCounter(0), counters.counter(TEST_COUNTER + i));
     }
 
-    counters.counter(TEST_COUNTER)++;
+    ++counters.counter(TEST_COUNTER);
     CPPUNIT_ASSERT_EQUAL(TCounter(1), counters.counter(TEST_COUNTER));
-    counters.counter(TEST_COUNTER)++;
+    ++counters.counter(TEST_COUNTER);
     CPPUNIT_ASSERT_EQUAL(TCounter(2), counters.counter(TEST_COUNTER));
-    counters.counter(TEST_COUNTER)--;
+    --counters.counter(TEST_COUNTER);
     CPPUNIT_ASSERT_EQUAL(TCounter(1), counters.counter(TEST_COUNTER));
-    counters.counter(TEST_COUNTER)--;
+    --counters.counter(TEST_COUNTER);
     CPPUNIT_ASSERT_EQUAL(TCounter(0), counters.counter(TEST_COUNTER));
 
     CProgramCountersTestRunner runners[N * 2];
-    for (int i = 0; i < N * 2; i++) {
+    for (int i = 0; i < N * 2; ++i) {
         runners[i].initialise(i, N);
     }
 
-    for (int i = 0; i < N * 2; i++) {
+    for (int i = 0; i < N * 2; ++i) {
         runners[i].start();
     }
 
-    for (int i = 0; i < N * 2; i++) {
+    for (int i = 0; i < N * 2; ++i) {
         runners[i].waitForFinish();
     }
 
     for (int i = 0; i < N; i++) {
         CPPUNIT_ASSERT_EQUAL(TCounter(0), counters.counter(TEST_COUNTER + i));
     }
 
-    for (int i = 0; i < 0x1000000; i++) {
-        counters.counter(TEST_COUNTER)++;
+    for (int i = 0; i < 0x1000000; ++i) {
+        ++counters.counter(TEST_COUNTER);
     }
     CPPUNIT_ASSERT_EQUAL(TCounter(0x1000000), counters.counter(TEST_COUNTER));
 }
 
+void CProgramCountersTest::testUnknownCounter() {
+    ml::core::CProgramCounters& counters = ml::core::CProgramCounters::instance();
+
+    const char* logFile = "test.log";
```

Comment:
as far as I remember there is some temporary directory to be used for unit tests: `ml::test::CTestTmpDir::tmpDir()`

Reply:
Thanks Hendrik. I'm not sure that making use of such a temporary directory is entirely feasible in this case. The logging configuration file would also require updating with the temporary path name which would only be possible by way of an environment variable. I think on the face of it that would be less attractive than the current solution.

There are a couple of other precedents setting up the logging in this way, see e.g. CAnomalyJobTest and PoissonMeanConjugateTest.

Reply:
nit: maybe you can add a code comment that `test.log` is linked to `testLogErrorsLog4cxx.properties` to document the reason for later